tests/db/test_v6_to_v7.py

```diff
@@ -0,0 +1,145 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import datetime
+import os
+import uuid
+from dataclasses import dataclass
+from typing import AsyncIterator, Awaitable, Callable, List
+
+import pytest
+from asyncpg import Connection, Record
+from asyncpg.cursor import Cursor
+
+from db.common import PGRestore
+from inmanta import data
+from inmanta.server.services.databaseservice import DatabaseService
+
+
+@pytest.fixture
+async def migrate_v6_to_v7(
+    hard_clean_db, hard_clean_db_post, postgresql_client: Connection, async_finalizer, server_config
+) -> AsyncIterator[Callable[[], Awaitable[None]]]:
+    """
+    Performs a v6 database restore and yields a function that migrates to v7 when called.
+
+    Word of caution:
+    This fixture differs from similar ones in previous version migration tests in the following ways:
+    - It yields a migration function instead of migrating before yielding. This to allow tests to set up desired preconditions.
```

Comment:
This should, I think, be set up in the dump of the previous version. So that would mean updating the test-case-data-generator, run it on the previous released code, put the dump here)

Reply:
You mean to make sure the dump contains these preconditions instead of extending them during the test run?

Reply:
yes

Reply:
27895b89 and 07950c2d